### ISSUE TITLE
Add filter/search box api support to custom tree views

### ISSUE THREAD

**Issue by eamodio** [microsoft/vscode#50062]
It would be great for an extension to provide the same filter/search box support that the new `Outline` view supports. Searching and filter custom views is a request I'm hearing more and more with GitLens.

/cc @jrieken

**Comment by Gruntfuggly** [comment]
This would be great, but it doesn't necessarily need to just be filter/search. There are other inputs which would be usefully located in the tree view - renaming, for example.

**Comment by ntotten** [comment]
This would be helpful for us on the salesforce test explorer as well. https://github.com/forcedotcom/salesforcedx-vscode/issues/605

**Comment by chnliquan** [comment]
does the custom treeProvider will provide a filter/search box? I think this requirement is very necessary, otherwise it is difficult to find the data you want in the case of a lot of data.
Is it possible to open the outline search box and filter it through the label property of the tree node?![image](https://user-images.githubusercontent.com/34906299/51294576-85702000-1a4e-11e9-8f78-d818571b97f9.png)
  @sandy081

**Comment by sandy081** [comment]
Might not be as an API but we are looking to implement a generic filtering for all trees which will just filter the data that is visible.

**Comment by chnliquan** [comment]
> Might not be as an API but we are looking to implement a generic filtering for all trees which will just filter the data that is visible.

thanks, looking forward to the implementation of the function, where can I get notice if the implementation is complete?

**Comment by cawoodm** [comment]
Is there any way to filter the explorer view today?
![image](https://user-images.githubusercontent.com/498834/54820872-d85d9200-4ca0-11e9-9e31-04e8d43cea93.png)

**Comment by mtxr** [comment]
I don't think so.

I'm looking forward to it as well.

**Comment by connorshea** [comment]
This'd be very useful 🙇

**Comment by superfliege** [comment]
any update on this?

**Comment by mtxr** [comment]
I'm looking forward to see it as 
well

**Comment by alekkowalczyk** [comment]
Coming from here https://github.com/hbenl/vscode-test-explorer/issues/4 
Would be really usefull and make the VSCode experience a step closed to the full VS :)

**Comment by usernamehw** [comment]
This issue most likely depends on #63566, which was repeatedly deferred from Dec 2018. It was in the Iteration Plan ... and it's already moved it into the `Deferred Items` #75103 this Iteration.

**Comment by eamodio** [comment]
Now that #63566 has landed, hopefully this can be supported!

**Comment by sandy081** [comment]
With new Tree, custom views can be filtered just like file explorer.

**Comment by usernamehw** [comment]
It doesn't search in nested folders which makes this mildly useless and inconsistent ... and it's not an API ... and not a real input. Not hating, just stating.

**Comment by sandy081** [comment]
It is consistent with other trees (file explorer). Exposing as an API is a big ASK which I am afraid might not happen soon.

**Comment by sandy081** [comment]
Closing this as the custom views with new tree supports filtering

**Comment by Freshchris01** [comment]
@sandy081 Is this feature somewhere documented? Couldn't find it..

**Comment by sandy081** [comment]
@alexr00 Is it documented in 1.37 release notes?

**Comment by alexr00** [comment]
It's called out [here](https://github.com/microsoft/vscode-docs/blob/master/release-notes/v1_37.md#new-tree-widget-adopted-in-tree-view) which links to more info and includes documentation about the Improved Keyboard Navigation.
